lib/resty/rediscluster.lua

```diff
@@ -21,13 +22,33 @@ local redis_crc = xmodem.redis_crc
 
 local DEFAULT_SHARED_DICT_NAME = "redis_cluster_slot_locks"
 local DEFAULT_REFRESH_DICT_NAME = "refresh_lock"
-local DEFAULT_MAX_REDIRECTION = 5
-local DEFAULT_MAX_CONNECTION_ATTEMPTS = 3
+local DEFAULT_MAX_REDIRECTION = 2
+local DEFAULT_MAX_CONNECTION_ATTEMPTS = 2
 local DEFAULT_KEEPALIVE_TIMEOUT = 55000
 local DEFAULT_KEEPALIVE_CONS = 1000
 local DEFAULT_CONNECTION_TIMEOUT = 1000
 local DEFAULT_SEND_TIMEOUT = 1000
 local DEFAULT_READ_TIMEOUT = 1000
+local DEFAULT_HEALTH_DICT_NAME = "redis_cluster_health"
+local inspect = require("inspect")
+
+local function health_check_timer()
+    local health_dict = ngx.shared[DEFAULT_HEALTH_DICT_NAME]
+    local all_keys = health_dict:get_keys()
+    for _, key in ipairs(all_keys) do
+        local failures = health_dict:get(key)
+        if failures > 3 then
+            -- Keep unhealthy nodes alive
+            health_dict:incr(key, 1)
+            health_dict:expire(key, 5)
+        else
+            -- Delete healthy nodes
+            health_dict:delete(key)
+        end
```

Comment:
What does this mean? You need to detect here whether the Redis service can connect normally (redis ping command).